docs/articles/documentation/guides/advanced-guides/speed-up-test-execution.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: docs
+title: Create Faster Tests
+permalink: /documentation/guides/advanced-guides/speed-up-test-execution.html
+---
+
+# Speed Up Test Execution
+
+This article describes ways to decrease the execution time of TestCafe tests. It includes the following sections:
+
+* [Run Tests Concurrently](#run-tests-concurrently)
+* [Run Tests in Headless Browsers](#run-tests-in-headless-browsers)
+* [Use Roles for Login](#use-roles-for-login)
+* [Set Test Speed](#set-test-speed)
+* [Run Tests in Local Browsers](#run-tests-in-local-browsers)
+* [Mock Requests](#mock-requests)
+* [Optimize Your Page Model](#optimize-your-page-model)
+
+## Run Tests Concurrently
+
+Enable concurrent mode to run multiple browser instances simultaneously. This speeds up the testing process at the expense of higher resource consumption.
+
+You can enable concurrency with the [concurrency](../../reference/configuration-file.md#concurrency) configuration file property:
+
+```json
+{
+  "concurrency": 3
+}
+```
+
+In CLI, use the [--concurency](../../reference/command-line-interface.md#-c-n---concurrency-n) option:
+
+```sh
+testcafe --concurency 3 chrome tests/
+```
+
+For more info on concurrency, read [Run Tests Concurrently](../../guides/basic-guides/run-tests.md#run-tests-concurrently).
+
+## Run Tests in Headless Browsers
+
+Headless browsers take less time to initialize because they lack a GUI. They also enable you to run tests in containers and other environments without graphical capabilities.
+
+For more information, see [Test in Headless Mode](../../guides/concepts/browsers.md#test-in-headless-mode)
+
+## Use Roles for Login
+
+Use [Roles](../../guides/advanced-guides/authentication.md#user-roles) to save time on user authentication.
+
+Roles are user log-in routines. When you first enable a `Role`, TestCafe executes the routine and saves the user authentication data that it receives. When you activate the same role later, TestCafe loads the authentication data and skips the log-in process.
+
+When you switch between `Roles`, TestCafe replaces your browser's existing authentication data with the new role's credentials.
+
+Place your `Role` statements inside a [beforeEach hook](../../reference/test-api/fixture/beforeeach.md) to recall the authentication data before each of the tests in your fixture.
+
+> Roles can access authentication data in cookie and browser storage only. If your authentication system stores data elsewhere, roles may not work.
+
+For more info on Roles, see [User Roles](../../guides/advanced-guides/authentication.md#user-roles)
+
+## Set Test Speed
+
+TestCafe emulates real user actions on tested webpages. Set the [speed](../../reference/command-line-interface.md#--speed-factor) option to change the emulation speed.
```

Comment:
> The minimum value of `0.01` represents a speed that is 100 times lower.

This coefficient affects the time between action events (for example, the `click` action is sequence of `mouseup`, `,mousedown`, `click`. So, we cannot say this.


What about such a variant?
```
TestCafe emulates the user actions with the maximum possible speed. It a default behavior and the `speed` value in this case equals `1`. 
Setting a lower `speed` can be useful for debugging, but it slows tests down.
If you set a custom`speed` value in the run configuration, disable this setting or set it to `1`.
```

Reply:
I'd say that `the time between action events` is roughly equal to `emulation speed` in this case. It's also an implementation detail.

We could  just cut out the `100 lower speed` part:

> TestCafe emulates real user actions on tested webpages. Set the [speed](../../reference/command-line-interface.md#--speed-factor) option to change the emulation speed.

> The highest value of `1` represents the fastest possible emulation speed. This is the default `speed` value. Setting a lower `speed` can be useful for debugging, but it slows tests down.

> If you set a custom `speed` value in the run configuration, disable this setting or set it to `1`.

Reply:
Ok. Then let's remove the `Set the speed option to change the emulation speed.` sentence from your variant. It's a call to action and we suggest to not use it.

Reply:
If we remove it, we lose information. How about we change it up a little?

> TestCafe emulates real user actions on tested webpages. The `speed` option controls the emulation speed.
>
> The highest value of 1 represents the fastest possible emulation speed. This is the default speed value. Setting a lower speed can be useful for debugging, but it slows tests down.
> If you set a custom speed value in the run configuration, disable this setting or set it to 1 to improve test speed.